### PR TITLE
upgrade: Check if Swift configuration is compatible with the upgrade

### PR DESCRIPTION
**Pull request by jsuchome**
**Why is this change necessary?**

Newton's Swift does not allow number of replicas bigger than number
of disks.

### CODE REVIEW

File: crowbar_framework/app/models/api/crowbar.rb [hunk 1]
```diff
@@ -171,6 +171,22 @@ def ceph_status
         ret
       end
 
+      def swift_check
+        # 1. number of replicas
+        ret = {}
+        prop = Proposal.where(barclamp: "swift").first
+        return ret if prop.nil?
+        replicas = prop["attributes"]["swift"]["replicas"]
```
Comment (dirkmueller) [inline crowbar_framework/app/models/api/crowbar.rb]: You need to multiply by the bumber of zones I thibk as well.

Comment (jsuchome) [inline crowbar_framework/app/models/api/crowbar.rb]: Are you sure? 
2 zones, 2 replicas and 2 swift-storage nodes works for me.
2 zones, 3 replicas, 2 swift-storage zones and I got an error before

Comment (jsuchome) [inline crowbar_framework/app/models/api/crowbar.rb]: https://github.com/openstack/swift/blob/master/swift/common/ring/builder.py#L437

**Comment by jsuchome** [comment]
@skazi0 new precheck -> crowbar-ui update

File: crowbar_framework/app/models/api/upgrade.rb [hunk 2]
```diff
@@ -362,6 +369,17 @@ def clusters_health_report_errors(check)
...
+      def swift_check_errors(check)
+        ret = {}
+        if check[:too_many_replicas]
+          ret[:too_many_replicas] = {
+            data: I18n.t("api.upgrade.prechecks.swift_replicas.error",
+              replicas: check[:too_many_replicas]),
+            help: I18n.t("api.upgrade.prechecks.swift_replicas.help")
+          }
+        end
+      end
```
Comment (skazi0) [inline crowbar_framework/app/models/api/upgrade.rb]: you miss one `ret` here

Comment (aplanas) [inline crowbar_framework/app/models/api/upgrade.rb]: indeed

**Comment by jsuchome** [comment]
Merging. @dirkmueller if you think this is wrong, we'll fix it in followup

**Comment by jsuchome** [comment]
[(2) swift issues: Add precheck for replica settings when upgrading from 6 to 7](https://trello.com/c/m6dpfsqO/2245-2-swift-issues-add-precheck-for-replica-settings-when-upgrading-from-6-to-7)

**Pull request merged**